javascripts/analytics.js

```diff
@@ -0,0 +1,90 @@
+function Analytics () {
+  this.$bodyElement = document.querySelector('body')
+  this.is404Page = this.$bodyElement.querySelector('h1') ? this.$bodyElement.querySelector('h1').textContent === 'Page not found' : false
+ }
+
+Analytics.prototype.init = function() {
+
+  this.$bodyElement.addEventListener('click', this.handleClickEvent.bind(this))
+
+  if (this.is404Page) {
+    this.page404Event()
+  }
+}
+
+Analytics.prototype.handleClickEvent = function (e) {
+  var target = e.target,
+      isLink = e.target.nodeName === 'A',
+      // external link: starts with http(s) or mailto and does not contain a "service.gov" in url
+      externalLinkRegex = /^(?=.*(http|mailto))(?:(?!cloud\.service\.gov\.uk).)*$/g,
```

Comment:
Now we're talking! Looks good to me :ok_hand: 

![Screenshot from 2021-04-28 14-50-07](https://user-images.githubusercontent.com/114097/116415282-2dc9dd00-a831-11eb-9d57-9fbcee6da3aa.png)